src/database_migration.py

```diff
@@ -0,0 +1,129 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Provide the DatabaseMigration class to manage database migrations."""
+import enum
+import logging
+from typing import cast
+
+import ops
+from ops.pebble import ExecError
+
+from charm_state import CharmState
+from constants import FLASK_APP_DIR, FLASK_STATE_DIR
+from exceptions import CharmConfigInvalidError
+
+logger = logging.getLogger(__name__)
+
+
+class DatabaseMigrationStatus(str, enum.Enum):
+    """Database migration status.
+
+    Attrs:
+        COMPLETED: one of the possible database migration status.
+        FAILED: one of the possible database migration status.
+        PENDING: one of the possible database migration status.
```

Comment:
```suggestion
        COMPLETED: A status denoting a successful database migration.
        FAILED: A status denoting an unsuccessful database migration.
        PENDING: A status denoting a pending database migration..
```
A bit of a trivial docstring, I think it can be good as is :D

Reply:
Updated, thanks!